src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+
+    public String getRandomColor() {
+        int index = random.nextInt(Colors.values().length);
+        Colors randomColors = Colors.values()[index];
+        return randomColors.toString();
+    }
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(Figures.values().length);
```

Comment:
Consider defining a constant for the number of figure types to avoid using magic numbers. For example, `private static final int FIGURE_COUNT = Figures.values().length;`